### PR TITLE
Fixed feature tests for Windows paths

### LINKED ISSUE #576: Tests do not pass on windows
Seems to mostly be path separator issues, it's close to passing fully, just needs a little fix.

```
$ npm test

> cucumber@0.10.2 test C:\dev\efokschaner\cucumber-js
> npm run lint && npm run unit-test && npm run feature-test


> cucumber@0.10.2 lint C:\dev\efokschaner\cucumber-js
> jshint bin features lib scripts spec


> cucumber@0.10.2 unit-test C:\dev\efokschaner\cucumber-js
> jasmine

Started
..............................................................................................................................................................................................................................................................................................................................................................................................................................................................................FFF..................FF................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

Failures:
1) Cucumber.Listener.RerunFormatter handleAfterFeaturesEvent() one failed scenario logs nothing
  Message:
    Expected 'path\to\scenario:1' to equal 'path/to/scenario:1'.
  Stack:
    Error: Expected 'path\to\scenario:1' to equal 'path/to/scenario:1'.
        at Object.<anonymous> (C:\dev\efokschaner\cucumber-js\spec\cucumber\listener\rerun_formatter_spec.js:40:42)

2) Cucumber.Listener.RerunFormatter handleAfterFeaturesEvent() two failed scenarios (same file) logs the path to the failed scenarios
  Message:
    Expected 'path\to\scenario:1:2' to equal 'path/to/scenario:1:2'.
  Stack:
    Error: Expected 'path\to\scenario:1:2' to equal 'path/to/scenario:1:2'.
        at Object.<anonymous> (C:\dev\efokschaner\cucumber-js\spec\cucumber\listener\rerun_formatter_spec.js:56:42)

3) Cucumber.Listener.RerunFormatter handleAfterFeaturesEvent() two failed scenarios (different file) logs the path to the failed scenarios
  Message:
    Expected 'path\to\scenario:1
    other\path\to\scenario:1' to equal 'path/to/scenario:1
    other/path/to/scenario:1'.
  Stack:
    Error: Expected 'path\to\scenario:1
    other\path\to\scenario:1' to equal 'path/to/scenario:1
    other/path/to/scenario:1'.
        at Object.<anonymous> (C:\dev\efokschaner\cucumber-js\spec\cucumber\listener\rerun_formatter_spec.js:72:42)

4) Cucumber.Listener.SummaryFormatter logFailures() when there is a failed step logs the failures
  Message:
    Expected spy log to have been called with [ '1) Scenario: scenarioName - path/to/scenario:1
       Step: stepKeyword stepName - path/to/step:2
       Step Definition: path/to/stepDefintion:3
       Message:
         failure exception stack

    ' ] but actual calls were [ 'Failures:

    ' ], [ '1) Scenario: scenarioName - path\to\scenario:1
       Step: stepKeyword stepName - path\to\step:2
       Step Definition: path\to\stepDefintion:3
       Message:
         failure exception stack

    ' ].
  Stack:
    Error: Expected spy log to have been called with [ '1) Scenario: scenarioName - path/to/scenario:1
       Step: stepKeyword stepName - path/to/step:2
       Step Definition: path/to/stepDefintion:3
       Message:
         failure exception stack

    ' ] but actual calls were [ 'Failures:

    ' ], [ '1) Scenario: scenarioName - path\to\scenario:1
       Step: stepKeyword stepName - path\to\step:2
       Step Definition: path\to\stepDefintion:3
       Message:
         failure exception stack

    ' ].
        at Object.<anonymous> (C:\dev\efokschaner\cucumber-js\spec\cucumber\listener\summary_formatter_spec.js:224:38)

5) Cucumber.Listener.SummaryFormatter logFailures() when there is an ambiguous step logs the failures
  Message:
    Expected spy log to have been called with [ '1) Scenario: scenarioName - path/to/scenario:1
       Step: stepKeyword stepName - path/to/step:2
       Message:
         Multiple step definitions match:
           pattern 1        - path/to/stepDefinition1:3
           longer pattern 2 - path/to/stepDefinition2:4

    ' ] but actual calls were [ 'Failures:

    ' ], [ '1) Scenario: scenarioName - path\to\scenario:1
       Step: stepKeyword stepName - path\to\step:2
       Message:
         Multiple step definitions match:
           pattern 1        - path\to\stepDefinition1:3
           longer pattern 2 - path\to\stepDefinition2:4

    ' ].
  Stack:
    Error: Expected spy log to have been called with [ '1) Scenario: scenarioName - path/to/scenario:1
       Step: stepKeyword stepName - path/to/step:2
       Message:
         Multiple step definitions match:
           pattern 1        - path/to/stepDefinition1:3
           longer pattern 2 - path/to/stepDefinition2:4

    ' ] but actual calls were [ 'Failures:

    ' ], [ '1) Scenario: scenarioName - path\to\scenario:1
       Step: stepKeyword stepName - path\to\step:2
       Message:
         Multiple step definitions match:
           pattern 1        - path\to\stepDefinition1:3
           longer pattern 2 - path\to\stepDefinition2:4

    ' ].
        at Object.<anonymous> (C:\dev\efokschaner\cucumber-js\spec\cucumber\listener\summary_formatter_spec.js:246:38)

965 specs, 5 failures
Finished in 2.14 seconds

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "unit-test"
npm ERR! node v0.12.4
npm ERR! npm  v2.10.1
npm ERR! code ELIFECYCLE
npm ERR! cucumber@0.10.2 unit-test: `jasmine`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the cucumber@0.10.2 unit-test script 'jasmine'.
npm ERR! This is most likely a problem with the cucumber package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     jasmine
npm ERR! You can get their info via:
npm ERR!     npm owner ls cucumber
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\dev\efokschaner\cucumber-js\npm-debug.log
npm ERR! Test failed.  See above for more details.

```

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Superset of #588 to get tests running on Travis.

Fixes feature tests on DOS with two changes:
* `features\step_definitions\cli_steps.js` runs `'node'` on executable .js files instead of the files directly
* `features\support\helpers.js` also normalizes slashes to the OS' `path.join('/')` result

Fixes #576.

### CODE REVIEW

File: features/support/helpers.js [hunk 1]
```diff
@@ -11,7 +13,8 @@ var helpers = {
       .replace(/^\s+/g, '')
       .replace(/\s+$/g, '')
       .replace(/[ \t]+\n/g, '\n')
-      .replace(/\d+m\d{2}\.\d{3}s/, '<duration-stat>');
+      .replace(/\d+m\d{2}\.\d{3}s/, '<duration-stat>')
+      .replace(/\//g, path.join('/'));
```
Comment (charlierudolph) [inline features/support/helpers.js]: Can `path.join('/')` just be path.sep? [docs](https://nodejs.org/api/path.html#path_path_sep)

File: spec/cucumber/listener/rerun_formatter_spec.js [hunk 2]
```diff
@@ -37,7 +38,7 @@ describe("Cucumber.Listener.RerunFormatter", function () {
       });
 
       it("logs nothing", function () {
-        expect(rerunFormatter.getLogs()).toEqual('path/to/scenario:1');
+        expect(rerunFormatter.getLogs()).toEqual(path.join('path/to/scenario:1'));
```
Comment (charlierudolph) [inline spec/cucumber/listener/rerun_formatter_spec.js]: unless we are actually passing multiple paths, I don't think using `path.join` makes sense. 
I would suggest using path.join like this:
```js
path.join('path', 'to', 'scenario:1')
```
Conversely we could just use `path.normalize` here. [docs](https://nodejs.org/api/path.html#path_path_normalize_path)

File: features/support/helpers.js (hunk 1 above)
Comment (JoshuaKGoldberg) [inline features/support/helpers.js]: Fixed!

Comment (JoshuaKGoldberg) [inline features/support/helpers.js]: Fixed!

File: package.json [hunk 3]
```diff
@@ -134,7 +134,7 @@
     "tmp": "0.0.28"
   },
   "scripts": {
-    "feature-test": "./bin/cucumber.js -p build",
+    "feature-test": "node ./bin/cucumber.js -p build",
     "feature-test-es5": "./bin/cucumber.js -p build -p es5",
```
Comment (charlierudolph) [inline package.json]: should this line also have `node` put in front of it?

Comment (JoshuaKGoldberg) [inline package.json]: Whoops, yes... Added.

File: package.json [hunk 4]
```diff
@@ -144,9 +144,9 @@
     "build-release": "node ./scripts/release.js"
   },
   "bin": {
-    "cucumber.js": "./bin/cucumber.js",
-    "cucumber-js": "./bin/cucumber.js",
-    "cucumberjs": "./bin/cucumber.js"
+    "cucumber.js": "node ./bin/cucumber.js",
+    "cucumber-js": "node ./bin/cucumber.js",
+    "cucumberjs": "node ./bin/cucumber.js"
```
Comment (charlierudolph) [inline package.json]: These should not be prefixed with node. npm creates symlinks based on these values so they need to stay as just paths

File: package.json [hunk 5]
```diff
@@ -134,8 +134,8 @@
     "tmp": "0.0.28"
   },
   "scripts": {
-    "feature-test": "./bin/cucumber.js -p build",
-    "feature-test-es5": "./bin/cucumber.js -p build -p es5",
+    "feature-test": "node ./bin/cucumber.js -p build",
```
Comment (charlierudolph) [inline package.json]: just checking in on this. So did running `npm run feature-test` not work on windows without this change?

File: package.json (hunk 4 above)
Comment (JoshuaKGoldberg) [inline package.json]: TIL, thanks.

File: package.json (hunk 5 above)
Comment (JoshuaKGoldberg) [inline package.json]: Correct. It parses the command as `'.'` and throws an error instead.

**Comment by charlierudolph** [comment]
Thanks!

**Comment by JoshuaKGoldberg** [comment]
Thanks for your patience :)

**Pull request merged**